python/dgl/_ffi/libinfo.py

```diff
@@ -47,30 +48,31 @@ def find_lib_path(name=None, search_path=None, optional=False):
 
     dll_path.append(install_lib_dir)
 
-    dll_path = [os.path.abspath(x) for x in dll_path]
     if search_path is not None:
-        if search_path is list:
-            dll_path = dll_path + search_path
-        else:
+        if isinstance(search_path, (list, tuple, set)):
+            dll_path = dll_path + list(search_path)
+        elif isinstance(search_path, str):
             dll_path.append(search_path)
-    if name is not None:
-        if isinstance(name, list):
-            lib_dll_path = []
-            for n in name:
-                lib_dll_path += [os.path.join(p, n) for p in dll_path]
         else:
-            lib_dll_path = [os.path.join(p, name) for p in dll_path]
-    else:
+            raise ValueError(f"type(search_path)={type(search_path)} is invalid")
```

Comment:
Format string is only available for python3.6+ while we currently support 3.5~3.8. Would you please change it to normal format syntax?